pkg/registry/chains/client/ns_client.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"net/url"
+
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/api/pkg/api/registry"
+
+	"github.com/networkservicemesh/sdk/pkg/registry/chains/connectto"
+	"github.com/networkservicemesh/sdk/pkg/registry/common/null"
+	"github.com/networkservicemesh/sdk/pkg/registry/common/serialize"
+	"github.com/networkservicemesh/sdk/pkg/registry/core/adapters"
+	"github.com/networkservicemesh/sdk/pkg/registry/core/chain"
+)
+
+// NSOption is an option pattern for NewNetworkServiceRegistryClient
+type NSOption func(nsOpts *nsClientOptions)
```

Comment:
Can we use one the option type for ns and nse stuff? It can prevent complex named functions such as `WithNSDialOptions` and `WithNSEDialOptions`

```suggestion
type Option func(opt *options)
type options struct {
nsFunctionality []registry.NetworkServiceRegistryClient
nseFunctionallity []registry.NetworkServiceEdnpointRegistryClient
dialOptions []grpc.DialOption
}
```

Reply:
Nice catch, done.